libs/vm_modules/tests/unit/string_tests.cpp

```diff
@@ -551,4 +551,86 @@ TEST_F(
   ASSERT_EQ(stdout.str(), "2 | aaa | ");
 }
 
+TEST_F(StringTests, utf8_split_returns_an_array_of_string_segments_with_the_separators_taken_out)
+{
+  static char const *TEXT = R"(
+    function main()
+      var text = '他身旁放着一支磨尖的花岗岩长矛备用，脚边卧着一头犬族猛兽，它发出的喘hōu声表明它虽已入睡却睡不安稳。';
+      var output = text.split('，');
```

Comment:
It may be worth using a non-ASCII character as the separator, like `着`. Another test above already checks that split works fine with an ASCII character.

Reply:
That is actually non-asci character (it just looks like normal ASCII coma & space,but in reality it is single utf-8 multi-byte character)

Reply:
But I agree, it would be better to use separator with more than single utf-8 multi-byte character.

Reply:
Maybe let's split this example string on `着一` so that someone who sees it in the future doesn't get confused by the no-quite-comma.